### PR TITLE
doc: update CHANGELOG for v1.18.0

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5060)

### CODE REVIEW

**Review by coryan — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5060)* status: all files reviewed, 7 unresolved discussions (waiting on @scotthart)

___
*[CHANGELOG.md, line 7 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5060#-MHIPlJR1FRusroST0y5:-MHIPlJR1FRusroST0y6:b-zabitj) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/d6ac0a202b4433b75e72df943841a3942d10a073/CHANGELOG.md#L7)):*
> ```Markdown
> ## v1.18.0 - 2020-09-15
> 
> ### Bigtable
> ```

Please add:

```
### Pub/Sub

The current release is a preview of the upcoming GA release. While we think the APIs are unlikely to change before GA, we reserve the right to change these APIs for now. This release may need optimization before it is ready for production workloads.
```
___
*[CHANGELOG.md, line 9 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5060#-MHIQEhF6gBqzjM8ilNm:-MHIQEhF6gBqzjM8ilNn:bhcgw8n) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/d6ac0a202b4433b75e72df943841a3942d10a073/CHANGELOG.md#L9)):*
> ```Markdown
> ### Bigtable
> 
> * fix: correct C++ version under MSVC (#5038)
> ```

Probably a dup?
___
*[CHANGELOG.md, line 18 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5060#-MHIQNCzCfq8yI50LTvd:-MHIQNCzCfq8yI50LTve:bii4waj) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/d6ac0a202b4433b75e72df943841a3942d10a073/CHANGELOG.md#L18)):*
> ```Markdown
> ### Spanner
> 
> * doc: adapt to new specification for NUMERIC samples (#5049)
> ```

Maybe say something like:

```
This release includes support for the `NUMERIC` data type in Cloud Spanner
```

or something to that effect?
___
*[CHANGELOG.md, line 19 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5060#-MHIPGQdDMGEtOqkLTHv:-MHIPGQdDMGEtOqkLTHw:b4uzka9) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/d6ac0a202b4433b75e72df943841a3942d10a073/CHANGELOG.md#L19)):*
> ```Markdown
> 
> * doc: adapt to new specification for NUMERIC samples (#5049)
> * doc: add "spanner_query_with_*_parameter" samples (#5016)
> ```

nit: `_` is a special character in markdown, this renders oddly, consider `spanner_query_with_*_parameter` with backticks.
___
*[CHANGELOG.md, line 24 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5060#-MHIPVBS5NEEPhoqnEmf:-MHIPVBS5NEEPhoqnEmg:bj79vjc) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/d6ac0a202b4433b75e72df943841a3942d10a073/CHANGELOG.md#L24)):*
> ```Markdown
> ### Common Libraries
> 
> * feat(common): cancel for internal::AsyncRetryLoop (#5045)
> ```

Probably skip?
___
*[CHANGELOG.md, line 28 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5060#-MHIPS2AEv2VMnLlLSoN:-MHIPS2AEv2VMnLlLSoO:b-2ghybk) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/d6ac0a202b4433b75e72df943841a3942d10a073/CHANGELOG.md#L28)):*
> ```Markdown
> * feat(common): make CompletionQueueImpl mockable [1] (#5036)
> * feat(common): make CompletionQueueImpl mockable [2] (#5039)
> * feat(common): make CompletionQueueImpl mockable [3] (#5043)
> ```

These three could be collapsed to one I think?
___
*[CHANGELOG.md, line 33 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5060#-MHIPbp1-PKwvkBSj7sf:-MHIPbp2-HXz4HBkgToK:b-peudnm) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/d6ac0a202b4433b75e72df943841a3942d10a073/CHANGELOG.md#L33)):*
> ```Markdown
> * fix: CompletionQueue shutdown disables RunAsync (#5008)
> * fix: remove 'Bigtable' from generic error message (#5034)
> * fix: remove unneeded dep (#5054)
> ```

What dep?  Maybe "remove unneded dep on absl::variant"?

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5060)* status: 0 of 1 files reviewed, 6 unresolved discussions (waiting on @coryan)

___
*[CHANGELOG.md, line 7 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5060#-MHIPlJR1FRusroST0y5:-MHIUCnk5TFNZ6d8Oc8l:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/d6ac0a202b4433b75e72df943841a3942d10a073/CHANGELOG.md#L7)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

Please add:

```
### Pub/Sub

The current release is a preview of the upcoming GA release. While we think the APIs are unlikely to change before GA, we reserve the right to change these APIs for now. This release may need optimization before it is ready for production workloads.
```

</blockquote></details>

Done.
___
*[CHANGELOG.md, line 9 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5060#-MHIQEhF6gBqzjM8ilNm:-MHIUDPv2RC12hFe3mUv:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/d6ac0a202b4433b75e72df943841a3942d10a073/CHANGELOG.md#L9)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

Probably a dup?
</blockquote></details>

Done.
___
*[CHANGELOG.md, line 18 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5060#-MHIQNCzCfq8yI50LTvd:-MHIUDwL2yoNeuQdTw1K:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/d6ac0a202b4433b75e72df943841a3942d10a073/CHANGELOG.md#L18)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

Maybe say something like:

```
This release includes support for the `NUMERIC` data type in Cloud Spanner
```

or something to that effect?
</blockquote></details>

Done.
___
*[CHANGELOG.md, line 24 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5060#-MHIPVBS5NEEPhoqnEmf:-MHIUEwS04EAJtI-G8G6:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/d6ac0a202b4433b75e72df943841a3942d10a073/CHANGELOG.md#L24)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

Probably skip?
</blockquote></details>

Done.
___
*[CHANGELOG.md, line 28 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5060#-MHIPS2AEv2VMnLlLSoN:-MHIUFMX8VwGmvucMRTr:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/d6ac0a202b4433b75e72df943841a3942d10a073/CHANGELOG.md#L28)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

These three could be collapsed to one I think?
</blockquote></details>

Done.
___
*[CHANGELOG.md, line 33 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5060#-MHIPbp1-PKwvkBSj7sf:-MHIUFoIDqGyuTN2y-sc:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/d6ac0a202b4433b75e72df943841a3942d10a073/CHANGELOG.md#L33)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

What dep?  Maybe "remove unneded dep on absl::variant"?
</blockquote></details>

Done.

**Review by coryan — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,34 @@
 # Changelog
 
-## v1.18.0 - TBD
+## v1.19.0 - TBD
+
+## v1.18.0 - 2020-09-15
```
Comment (devbww) [inline CHANGELOG.md:5]: We've never added a day number before, even when having multiple releases in the same month.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5060)* status: 0 of 1 files reviewed, 7 unresolved discussions (waiting on @coryan and @devbww)

___
*[CHANGELOG.md, line 5 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5060#-MHIOrrEa_pMFhX_v-Wm-r2-5:-MHIkokW92oFxq79ovRs:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/1860864a3bc79850069b541c948414ef798eefeb/CHANGELOG.md#L5)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

We've never added a day number before, even when having multiple releases in the same month.
</blockquote></details>

Done.

**Pull request merged**